lib/utils/ensure-platform-support.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { parse as parseCookie } from 'cookie';
+
+import BootWarning from '../components/boot-warning';
+
+const hasLocalStorage = () => {
+  try {
+    localStorage.setItem('__localStorageSentinel__', 'present');
+    localStorage.removeItem('__localStorageSentinel__');
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+const hasIndexedDB = () => {
+  try {
+    const opener = indexedDB.open('simplenote_sentinel');
+    if (!(opener instanceof IDBOpenDBRequest)) {
+      return false;
+    }
+    const closer = () => indexedDB.deleteDatabase('simplenote_sentinel');
+    opener.onerror = closer;
+    opener.onsuccess = closer;
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+const hasCookies = () => {
+  try {
+    if (!navigator.cookieEnabled) {
+      return false;
+    }
+
+    const cookie = document.cookie;
+
+    const now = Date.now().toString();
+    document.cookie = `__now=${now};`;
+    const parsed = parseCookie(document.cookie);
+    const didSet = parsed.__now === now;
+    document.cookie = cookie;
+
+    return didSet;
+  } catch (e) {
+    return false;
+  }
+};
+
+const deps = [
+  ['localStorage', hasLocalStorage()],
+  ['indexedDB', hasIndexedDB()],
+  ['cookies', hasCookies()],
+];
+
+const missingDeps = deps.filter(([, hasIt]) => !hasIt).map(([name]) => name);
+
+if (missingDeps.length) {
+  ReactDOM.render(
+    <BootWarning>
+      <p>
+        Simplenote depends on a few web technologies to operate. Please make
+        sure that you have all of the following enabled in your browser.
```

Comment:
@SylvesterWilmott would you be willing to provide some copy-review on this?

This message is the only thing that would appear to a customer if they try to load Simplenote without having the necessary support enabled in the browser.

Reply:
The copy looks good to me! Thanks @dmsnell